src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user = userService.findByLogin(login).orElseThrow(()
+                -> new AuthenticationException("This login: " + login + " doesn't exist!"));
+        if (user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt()))) {
+            return user;
+        }
+        throw new AuthenticationException("Wrong password!");
+    }

```

Comment:
To avoid brute force we don’t have to give a specific hint that of the data entered is incorrect. Also we have a rule 
![image](https://github.com/mate-academy/hibernate-user-service-hw/assets/73792778/a00003fc-7c04-4232-9db2-41547143bd98)
```suggestion
    @Override
    public User login(String login, String password) throws AuthenticationException {
        User user = userService.findByLogin(login).orElseThrow(()
                -> new AuthenticationException("This login: " + login + " doesn't exist!"));
        if (user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt()))) {
            return user;
        }
        throw new AuthenticationException("Wrong password!");
    }

```